adaptor/Dockerfile

```diff
@@ -0,0 +1,14 @@
+ARG BUILDPLATFORM=linux/amd64
+
+FROM --platform=$BUILDPLATFORM golang:1.16.4-alpine as golang
+WORKDIR /linkerd-smi-build
+COPY go.mod go.sum ./
+RUN go mod download
+COPY adaptor adaptor
+COPY pkg pkg
+RUN CGO_ENABLED=0 GOOS=linux go build -o /out/smi-adaptor -tags production -mod=readonly -ldflags "-s -w" ./adaptor
```

Comment:
I commented earlier about `TARGETARCH` not being used int the former `go-deps` stage. But it's still required in this golang stage in order to support multiarch builds through the `GOARCH` env var.

Reply:
Ahh, Right. I was in the impression that we can add it later but seems too simple to add it now!